### ISSUE TITLE
[Feature Request] Add check for dead links in the documentation

### ISSUE THREAD

**Issue by jlengrand** [open-wc/open-wc#2060]
I created a simple workflow so that we have an overview of dead links in the documentation when pull requests and releases are created.

I found some dead links in the past and there are likely more. This would help spot and fix them :).

**Comment by jlengrand** [comment]
See https://github.com/open-wc/open-wc/pull/2059 for proposal

**Comment by daKmoR** [comment]
the check is already done in the task before via `yarn site:build` which executes Rocket... which has `check-html-links` out of the box...

If you open the "build" phase you will see it executed
```
👀 Checking if all internal links work...
✅ All internal links are valid.
Done in 88.22s.
```

the reason there are still some broken links is https://github.com/open-wc/open-wc/issues/2054

**Comment by jlengrand** [comment]
Right, I checked but didn't look deep enough. Some links are broken, but the build passes though in your screenshot?
